src/errors.rs

```diff
@@ -91,8 +91,11 @@ impl DockerErrorDetails for RepoError {
     }
     fn docker_message(&self, info: &FetchInfo) -> String {
         match self {
-            RepoError::Git(git2::ErrorCode::NotFound) => format!("git ref: {} not found", &info.reference),
-            RepoError::Git(_) => "unknown git error".to_string(),
+            RepoError::Git(e) => if e.code() == git2::ErrorCode::NotFound {
+                format!("git ref: {} not found", &info.reference)
+            } else {
+                "unknown git error".to_string()
```

Comment:
Why are we not doing `e.to_string()` here or something to always get whatever we can?

Reply:
Generally because we might want a difference between what we log and what we emit to the client. This message is what's returned to the docker client. I don't know if there might be cases where the error message contains sensitive info. 

But I'm fine with changing this. :) I just think logging full errors is more important than returning them through the API, especially when I'm not sure what they might contain.

Reply:
Oh, are we logging the full errors? I didn't think we were.. Can you show me where? If so, I'm happy with this and think your concerns are reasonable!

Reply:
> Oh, are we logging the full errors?

We are in some cases at least. At least with the error at hand recently ("git proxy options on fetch")

Not sure about full coverage of log statements for all operations though, but will check in a minute... In any case: "fn docker_message()" seems like the wrong place to log :)